collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_user.py

```diff
@@ -199,16 +199,19 @@ def create_user(module, array):
                             module.params["name"]
                         )
                     )
-            if module.params["role"] != user["role"]:
-                try:
-                    array.set_admin(module.params["name"], role=module.params["role"])
-                    role_changed = True
-                except Exception:
-                    module.fail_json(
-                        msg="Local User {0}: Role changed failed".format(
-                            module.params["name"]
+            if module.params["role"] and module.params["role"] != user["role"]:
+                if module.params["name"] != "pureuser":
```

Comment:
Should it not return a warning/notice when trying to change `pureuser`? Right now it just returns `ok`.

Reply:
@dnix101 warning message added